server/src/lib/model/db.ts

```diff
@@ -678,4 +679,28 @@ export default class DB {
     );
     return row;
   }
+
+  async hasChallengeEnded(challenge: ChallengeToken) {
+    let hasChallengeEnded = true;
```

Comment:
We probably should name these different things. Maybe the method could be called `getHasChallengeEnded()`?

Reply:
OK

Reply:
maybe the variable could be called challengeEnded.